app/controllers/submissions_controller.rb

```diff
@@ -2,21 +2,21 @@ class SubmissionsController < ApplicationController
   before_action :set_model, only: [ :show, :update ]
 
   def index
-    @models = Current.user.models
+    @models = policy_scope(Model).where(owner: Current.user)
```

Comment:
We can access current user in the policy scope:

```suggestion
    @models = policy_scope(Model)
```